ffv1.md

```diff
@@ -479,6 +479,26 @@ RFC:```
 RFC:j_{0} = 2
 RFC:```
 
+##### Termination
+
+The range coder can be used in 3 modes.
+
+--------------- ----------------------------------------------------------------
+`Open mode`     In which when decoding, every symbol the reader attempts to read is available. In this mode arbitrary data can have been appended without affecting the range coder output. This mode is not used in FFv1.
+
+`Closed mode`   In which the length in bytes of the bytestream is provided to the range decoder. Bytes beyond the length are read as 0 by the range decoder. This is generally 1 byte shorter than the open mode.
+
+`Sentinel mode` In which case the exact length in bytes is not known. Thus the range decoder may read into non range coded data following range coded data. In this mode after all range coded symbols one binary symbol with state 129 is read and the value discarded. After reading this symbol the range decoder will have read one byte beyond the end of the range coded bytestream. This way the byte position of the end can be determined. Bytestreams written in sentinel mode can be read in Closed mode if the length can be determined, in this case the last (sentinel) symbol will be read non corrupted and be of value 0.
+--------------- ----------------------------------------------------------------
+
+Above describes the range decoding, encoding is defined as any process which produces a decodable bytestream.
+
+There are 3 places where range coder termination is needed in FFv1.
+First is in the ConfigurationRecord, in this case the size of the range coded bytestream is known, this is handled as Closed mode.
+Second is the switch from the slice header which is range coded to golomb coded slices. This is handled as Sentinel mode.
+Third is the end of range coded slices which need to terminate before the CRC at their end. This can be handled as Sentinel mode
+or as Closed mode if the CRC position has been determined.
```

Comment:
rewrite to make references consistent:

"First is in the \`Configuration Record\`, in this case the size of the range coded bytestream is known and handled as \`Closed mode\`.
Second is the switch from the \`Slice Header\` which is range coded to Golomb coded slices as \`Sentinel mode\`.
Third is the end of range coded Slices which need to terminate before the CRC at their end. This can be handled as \`Sentinel mode\` or as \`Closed mode\` if the CRC position has been determined."